src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -3572,5 +3572,68 @@ public bool TryGetSimilarName(DName name, FieldNameKind fieldNameKind, out strin
             return similar != null &&
                    comparer.Distance(similar) < (name.Value.Length / 3) + 3;
         }
+
+        /// <summary>
+        /// Try to union all table child types and checks if any coercion is necessary. Meant to be called from within table type check loop.
```

Comment:
> Try to union all table child types and checks if any coercion is necessary [](http://example.com/codeflow?start=11&length=75)

Type1 - {A:{x:{d1:1,d2:3}, y:3}} 
Type2 - {A:{x:{d1:"1", d2:4}, y:3}}

Type1.TryUnionWithCoerce

Does this handle nested types like this?